### PR TITLE
Catch and report session scope fixture teardown failures in a separated dedicated test case

### PR DESCRIPTION
**Pull request by ebenahar**
Partially fixes 4612

Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

File: tests/test_failure_propagator.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import pytest
+
+from ocs_ci.framework.pytest_customization.marks import tier1
+
+
+@tier1
+@pytest.mark.last
```
Comment (AviadP) [inline tests/test_failure_propagator.py:7]: We already marked one test as last already: https://github.com/red-hat-storage/ocs-ci/blob/66e42efb8b150e4e3d86467a166b5db1ddd2b419/tests/manage/z_cluster/cluster_expansion/test_add_capacity.py#L88, need to change it's order to be `second_to_last`

Comment (ebenahar) [reply tests/test_failure_propagator.py:7]: great catch

**Review by AviadP — APPROVED** [review]

**Comment by ebenahar** [comment]
The failure in the verification run is not related and will be fixed with https://github.com/red-hat-storage/ocs-ci/pull/4623

**Review by dahorak — APPROVED** [review]

File: tests/test_failure_propagator.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class TestFailurePropagator:
+    """
+    Test class for failure propagator test case. The test intention is to run last and propagate
```
Comment (vasukulkarni) [inline tests/test_failure_propagator.py:10]: second to last?

Comment (ebenahar) [reply tests/test_failure_propagator.py:10]: this one runs last

**Pull request merged**